From 63ca21fddaf4f3b12baf320857407b884db2b570 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 02 Jul 2020 19:20:33 +0000
Subject: [PATCH] GCC: fix operator<< for performance_manager::PageNode::OpenedType

Move operator<< into namespace performance_manager.

Bug: 819294
Change-Id: I4e0e97f10dca0b2e56308e07a67cb00e2c2e8052
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2273191
Reviewed-by: Chris Hamilton <chrisha@chromium.org>
Commit-Queue: Chris Hamilton <chrisha@chromium.org>
Cr-Commit-Position: refs/heads/master@{#784897}
---

diff --git a/components/performance_manager/graph/page_node.cc b/components/performance_manager/graph/page_node.cc
index 113fcf4..b1a4007 100644
--- a/components/performance_manager/graph/page_node.cc
+++ b/components/performance_manager/graph/page_node.cc
@@ -32,11 +32,11 @@
 PageNode::ObserverDefaultImpl::ObserverDefaultImpl() = default;
 PageNode::ObserverDefaultImpl::~ObserverDefaultImpl() = default;
 
-}  // namespace performance_manager
-
 std::ostream& operator<<(
     std::ostream& os,
     performance_manager::PageNode::OpenedType opened_type) {
   os << performance_manager::PageNode::ToString(opened_type);
   return os;
-}
\ No newline at end of file
+}
+
+}  // namespace performance_manager
diff --git a/components/performance_manager/public/graph/page_node.h b/components/performance_manager/public/graph/page_node.h
index 422c043..79fec03 100644
--- a/components/performance_manager/public/graph/page_node.h
+++ b/components/performance_manager/public/graph/page_node.h
@@ -265,10 +265,10 @@
   DISALLOW_COPY_AND_ASSIGN(ObserverDefaultImpl);
 };
 
-}  // namespace performance_manager
-
 // std::ostream support for PageNode::OpenedType.
 std::ostream& operator<<(std::ostream& os,
                          performance_manager::PageNode::OpenedType opened_type);
 
+}  // namespace performance_manager
+
 #endif  // COMPONENTS_PERFORMANCE_MANAGER_PUBLIC_GRAPH_PAGE_NODE_H_
